R/rcpp.R

```diff
@@ -6,28 +6,20 @@
 #'   * May create an initial placeholder `.c` or `.cpp` file
 #'   * Creates `Makevars` and `Makevars.win` files (`use_rcpp_armadillo()` only)
 #'
-#' @param name If supplied, creates and opens `src/name.{c,cpp}`.
-#'
-#' @details
-#'
-#' When using compiled code, please note that there must be at least one file
-#' inside the `src/` directory prior to building the package. As a result,
-#' if an empty `src/` directory is detected, either a `.c` or `.cpp` file will
-#' be added.
-#'
+#' @inheritParams use_r
 #' @export
 use_rcpp <- function(name = NULL) {
   check_is_package("use_rcpp()")
   check_uses_roxygen("use_rcpp()")
-  check_not_empty_file_name(name)
-
-  use_src()
 
   use_dependency("Rcpp", "LinkingTo")
   use_dependency("Rcpp", "Imports")
   roxygen_ns_append("@importFrom Rcpp sourceCpp") && roxygen_remind()
 
-  use_src_example_script(name, "cpp")
+  use_src()
+  path <- path("src", compute_name(name, "cpp"))
+  use_template("code.cpp", path)
+  edit_file(path)
```

Comment:
More bad `proj_path()` vibes.

Reply:
Actually, is there a reason we aren't just specifying `open = is_interactive()` in `use_template()`?

Reply:
Yeah, because that only opens the file if it changed.